src/function/table/arrow/arrow_duck_schema.cpp

```diff
@@ -52,6 +56,251 @@ void ArrowType::ThrowIfInvalid() const {
+	} else if (format == "tDn") {
+		return make_shared_ptr<ArrowType>(LogicalType::INTERVAL,
+		                                  make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::NANOSECONDS));
+	} else if (format == "tiD") {
+		return make_shared_ptr<ArrowType>(LogicalType::INTERVAL, make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::DAYS));
+	} else if (format == "tiM") {
+		return make_shared_ptr<ArrowType>(LogicalType::INTERVAL,
+		                                  make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::MONTHS));
+	} else if (format == "tin") {
+		return make_shared_ptr<ArrowType>(LogicalType::INTERVAL,
+		                                  make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::MONTH_DAY_NANO));
+	} else if (format == "z") {
+		auto type_info = make_uniq<ArrowStringInfo>(ArrowVariableSizeType::NORMAL);
+		return make_shared_ptr<ArrowType>(LogicalType::BLOB, std::move(type_info));
+	} else if (format == "Z") {
+		auto type_info = make_uniq<ArrowStringInfo>(ArrowVariableSizeType::SUPER_SIZE);
+		return make_shared_ptr<ArrowType>(LogicalType::BLOB, std::move(type_info));
+	} else if (format[0] == 'w') {
+		string parameters = format.substr(format.find(':') + 1);
+		auto fixed_size = NumericCast<idx_t>(std::stoi(parameters));
+		auto type_info = make_uniq<ArrowStringInfo>(fixed_size);
+		return make_shared_ptr<ArrowType>(LogicalType::BLOB, std::move(type_info));
+	} else if (format[0] == 't' && format[1] == 's') {
+		// Timestamp with Timezone
+		// TODO right now we just get the UTC value. We probably want to support this properly in the future
+		unique_ptr<ArrowTypeInfo> type_info;
+		if (format[2] == 'n') {
+			type_info = make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::NANOSECONDS);
+		} else if (format[2] == 'u') {
+			type_info = make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::MICROSECONDS);
+		} else if (format[2] == 'm') {
+			type_info = make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::MILLISECONDS);
+		} else if (format[2] == 's') {
+			type_info = make_uniq<ArrowDateTimeInfo>(ArrowDateTimeType::SECONDS);
+		} else {
+			throw NotImplementedException(" Timestamptz precision of not accepted");
+		}
+		return make_shared_ptr<ArrowType>(LogicalType::TIMESTAMP_TZ, std::move(type_info));
+	}
+
+	return nullptr;
+}
+
+shared_ptr<ArrowType> ArrowType::CreateListType(DBConfig &config, ArrowSchema &child, ArrowVariableSizeType size_type,
+                                                bool view) {
+	auto child_type = GetArrowLogicalType(config, child);
+
+	unique_ptr<ArrowTypeInfo> type_info;
+	auto type = LogicalType::LIST(child_type->GetDuckType());
+	if (view) {
+		type_info = ArrowListInfo::ListView(std::move(child_type), size_type);
+	} else {
+		type_info = ArrowListInfo::List(std::move(child_type), size_type);
+	}
+	return make_uniq<ArrowType>(type, std::move(type_info));
+}
+
+shared_ptr<ArrowType> ArrowType::GetTypeFromFormatNested(DBConfig &config, ArrowSchema &schema, string &format) {
```

Comment:
As we discussed, this can be merged back into `GetTypeFromFormat` I think, there ended up being no need to separate non-nested from nested types